### ISSUE TITLE
Let `--exec` print final commands?

### ISSUE THREAD

**Issue by hg-zt** [sharkdp/fd#943]
It is helpful for debugging and showing progress

The idea is
```bash
$ fd .mp4 --threads=1 --exec-show-command some-video-processing-program --input {} --output {.}-output.mp4 
> some-video-processing-program --input 1.mp4 --output 1-output.mp4
(...actual output of the program)
> some-video-processing-program --input 2.mp4 --output 2-output.mp4
> some-video-processing-program --input subdir/1.mp4 --output subdir/1-output.mp4
```
It's like `set -x` for bash.

**Comment by sharkdp** [comment]
Thank you for the feature request. Did you see the discussion in #946? I think it would be good to think about this some more before implementing it. One idea to support this would be to implement #406 (see discussion therein)

**Comment by tmccombs** [comment]
Some other options (mentioned in the PR):

- Add something to documentation mentioning you can put echo in front of the command to get a dry run with warning that it may not be the exact commands that are actually run. 
- add an equivalent of the -ok option in find that asks the user to confirm before actually running each command. Name could be interactive, confirm, verify, etc

**Comment by sharkdp** [comment]
@tmccombs Note that this feature request is slightly different, as it asks for the printing of the command just before it is being executed (not a dry run). So the `echo` approach currently doesn't work. But it would work if we allowed for multiple `-x`/`--exec` instances.

**Comment by hg-zt** [comment]
> Thank you for the feature request. Did you see the discussion in #946? I think it would be good to think about this some more before implementing it. One idea to support this would be to implement #406 (see discussion therein)

It is somehow different from `--dry-run` because one of my use cases is to show the progress for long-running jobs, i.e. which file/directory is under processing. Currently I'm using `-x bash -c` as a workaround.
And yeah I think the request could be solved by implementing #406.

**Comment by Flimm** [comment]
This issue was closed as completed, but I'm confused. What is the fix? How can I make `fd` print the command before running it?

The only solution I found was to resort to using `xargs`:

    fd -0 --glob '*.mp4' | xargs -0 -I {} sh -c 'echo {} ; some-video-processing-program {}'
